### PR TITLE
Pull parts of FirebaseAuth into an abstract class.

### PR DESCRIPTION
**Pull request by micahstairs**
This pull request moves part of FirebaseAuth into its abstract class as part of adding multi-tenancy support (see issue #332).

This class will eventually have a getTenantManager() method.

### CODE REVIEW

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 1]
```diff
@@ -53,35 +33,19 @@
...
-    this.cookieVerifier = threadSafeMemoize(builder.cookieVerifier);
-    this.userManager = threadSafeMemoize(new Supplier<FirebaseUserManager>() {
-      @Override
-      public FirebaseUserManager get() {
-        return new FirebaseUserManager(firebaseApp);
-      }
-    });
-    this.jsonFactory = firebaseApp.getOptions().getJsonFactory();
+    super(
+        checkNotNull(builder.firebaseApp),
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Do `checkNotNull` in the parent constructor.

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 2]
```diff
@@ -1207,4 +148,4 @@ public void destroy() {
       instance.destroy();
     }
   }
-}
+}
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Add newline

Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Done.

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java (hunk 1 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Done.

**Comment by hiranya911** [comment]
I think this PR mainly just moves the existing code/docs around. There's probably no need to review docs here.

**Pull request merged**